snoop/data/models.py

```diff
@@ -425,6 +426,42 @@ def delete(self, *args, **kwargs):
         digests.retry_index(self.blob)
 
 
+class EntityType(models.Model):
+    type = models.CharField(max_length=256, unique=True)
```

Comment:
Are we sure we want to implement custom entity types created by the user in this way? Without some kind of training phase and/or predictive coding, this seems kind of useless and will only complicate the code. Sure, you can have a rule to re-tag previously set "NAZI" type entities with the same custom type, but it won't follow the context: if the same text for your NAZI entity is used in some other context (where it's the name of a place or a work of art or something else), then you'll have a wrong match.

Maybe what we want here is for people to be able to tag some existing (entity, type) pair with their own tags, so they'd say `PERSON: Adolf Hitler` has the "NAZI" tag - and we'd tag the whole document with `entity-custom-tags: ['NAZI']`


If we do it that way, then we don't want a whole table for this when it's only used as a Foreign Key on the Entity. It's going to be annoying to manage this from python code; you have to add more data migrations to append values to this list.

Let's use a multiple choice string field on the `Entity.type` (same as in `snoop.data.models.Tasks.status`) and remove this table and its data migration, then add another table called "CustomEntityType" or "CustomEntityTag" or something, with a Many to Many relationship with "Entity"

Reply:
Custom entity types are pretty important, but I don't think that we'll need something as complicated as predicting these custom entity types. It's enough to just match the ones ppl add to the list of entities of their custom type entity list documents. Since ppl need to be able to add entities to existing types as well, there's not much of a difference.
Let's say you have a drug trafficking ring or something who operates with animal names as code names. First, it would be helpful to have a custom entity type `RING` or something, and maybe you want to consider `owl` as an entity in this context. 
We should be able to look for these custom entities just like for the others.

Reply:
> going to be annoying to manage this from python code; 
> you have to add more data migrations to append values to this list.

Wait, I don't really get that. Why do we need new migrations if we add to the list? Can't we just use `EntityType.objects.get_or_create(type='NAZI')` as we do with all other parts of the model?

Reply:
Ok, then let's do `EntityType.objects.get_or_create` everywhere and remove the migration that adds the rows with classes. The classes will be added when they appear in the data, and we won't have any "magic" classes that come pre-created in the database.